### PR TITLE
Proxy streaming when using async client 18.12

### PR DESCRIPTION
**Pull request by cmangeat**
see https://github.com/georchestra/georchestra/pull/2547.

### CODE REVIEW

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java [hunk 1]
```diff
@@ -787,7 +738,58 @@ private Header extractHeaderSetCookie(HttpResponse proxiedResponse) {
...
+                HttpEntity streamEntity = new InputStreamEntity(pis, contentType);
+                httpResponse.setEntity(streamEntity);
+                future.complete(httpResponse);
+            }
+
+            protected void onContentReceived(ContentDecoder decoder, IOControl ioctrl) throws IOException {
+                int bytesRead = decoder.read(this.bbuf);
+                if (bytesRead > 0) {
+                    this.bbuf.flip();
+                    channel.write(this.bbuf);
```
Comment (groldan) [inline security-proxy/src/main/java/org/georchestra/security/Proxy.java]: Beware the output channel have written less than bbuf.remaining bytes, I guess we should make it something like ```while(this.bbuf.hasRemaining()) {
                    	channel.write(this.bbuf);
                    }```

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java [hunk 2]
```diff
@@ -787,7 +738,58 @@ private Header extractHeaderSetCookie(HttpResponse proxiedResponse) {
 
     @VisibleForTesting
     protected HttpResponse executeHttpRequest(CloseableHttpAsyncClient httpclient, HttpRequestBase proxyingRequest) throws IOException, TimeoutException, ExecutionException, InterruptedException {
-        Future<HttpResponse> future = httpclient.execute(proxyingRequest, null);
+        CompletableFuture<HttpResponse> future = new CompletableFuture<HttpResponse>();
+
+        PipedOutputStream pos = new PipedOutputStream();
```
Comment (groldan) [inline security-proxy/src/main/java/org/georchestra/security/Proxy.java]: Mape `pos`, `pis`, and `channel` members of the `consumer` inner class? Also add `@Override` annotations to `onEntityEnclosed`, `onContentReceived`, and `releaseResources`?

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java [hunk 3]
```diff
@@ -251,16 +260,13 @@ public void init() throws Exception {
 
         // Proxy permissions not set by datadir
         if (proxyPermissionsFile != null && proxyPermissions == null) {
-            Closer closer = Closer.create();
-            try {
+            try (Closer closer = Closer.create()) {
```
[the remark this answers is not in the document]
Comment (cmangeat) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java]: thank you !

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java (hunk 1 above)
Comment (cmangeat) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java]: thank you !

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java (hunk 2 above)
Comment (cmangeat) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java]: thank you !

**Pull request merged**